triton-vm/src/table/master_table.rs

```diff
@@ -1046,59 +1048,117 @@ pub fn all_quotients(
     let tran_section_end = cons_section_end + MasterExtTable::NUM_TRANSITION_CONSTRAINTS;
-    let term_section_end = tran_section_end + MasterExtTable::NUM_TERMINAL_CONSTRAINTS;
-
-    prof_start!(maybe_profiler, "initial", "AIR");
-    MasterExtTable::fill_initial_quotients(
-        quotient_domain_master_base_table,
-        quotient_domain_master_ext_table,
-        &mut quotient_table.slice_mut(s![.., ..init_section_end]),
-        initial_quotient_zerofier_inverse(quotient_domain).view(),
-        challenges,
-    );
-    prof_stop!(maybe_profiler, "initial");
-
-    prof_start!(maybe_profiler, "consistency", "AIR");
-    MasterExtTable::fill_consistency_quotients(
-        quotient_domain_master_base_table,
-        quotient_domain_master_ext_table,
-        &mut quotient_table.slice_mut(s![.., init_section_end..cons_section_end]),
-        consistency_quotient_zerofier_inverse(trace_domain, quotient_domain).view(),
-        challenges,
-    );
-    prof_stop!(maybe_profiler, "consistency");
-
-    prof_start!(maybe_profiler, "transition", "AIR");
-    MasterExtTable::fill_transition_quotients(
-        quotient_domain_master_base_table,
-        quotient_domain_master_ext_table,
-        &mut quotient_table.slice_mut(s![.., cons_section_end..tran_section_end]),
-        transition_quotient_zerofier_inverse(trace_domain, quotient_domain).view(),
-        challenges,
-        trace_domain,
-        quotient_domain,
+
+    prof_start!(maybe_profiler, "zerofier inverse");
+    let initial_zerofier_inverse = initial_quotient_zerofier_inverse(quotient_domain);
+    let consistency_zerofier_inverse =
+        consistency_quotient_zerofier_inverse(trace_domain, quotient_domain);
+    let transition_zerofier_inverse =
+        transition_quotient_zerofier_inverse(trace_domain, quotient_domain);
+    let terminal_zerofier_inverse =
+        terminal_quotient_zerofier_inverse(trace_domain, quotient_domain);
+    prof_stop!(maybe_profiler, "zerofier inverse");
+
+    prof_start!(
+        maybe_profiler,
+        "evaluate AIR / divide zerofiers / combine quotients"
     );
-    prof_stop!(maybe_profiler, "transition");
-
-    prof_start!(maybe_profiler, "terminal", "AIR");
-    MasterExtTable::fill_terminal_quotients(
-        quotient_domain_master_base_table,
-        quotient_domain_master_ext_table,
-        &mut quotient_table.slice_mut(s![.., tran_section_end..term_section_end]),
-        terminal_quotient_zerofier_inverse(trace_domain, quotient_domain).view(),
-        challenges,
+    let quotient_codeword = (0..quotient_domain.length)
+        .into_par_iter()
+        .map(|row| {
+            // get views of relevant rows
+            let unit_distance = quotient_domain.length / trace_domain.length;
+            let domain_length_bit_mask = quotient_domain.length - 1;
+            let next_row_index = (row + unit_distance) & domain_length_bit_mask;
```

Comment:
I think we can afford a modulo operation.
```suggestion
            let next_row_index = (row + unit_distance) % quotient_domain.length;
```
Knowing that the snippets are equivalent is not necessarily obvious. I'm assuming that the applied optimiziation is not measurable.